lib/rules/lint-table-groups.js

```diff
@@ -1,30 +1,22 @@
 'use strict';
 
+const AstNodeInfo = require('../helpers/ast-node-info');
 const Rule = require('./base');
 const message = 'Tables must have a table group (thead, tbody or tfoot).';
 
 const ALLOWED_TABLE_CHILDREN = ['thead', 'tbody', 'tfoot', 'caption', 'colgroup'];
 
-function hasDisallowedChildren(tableNode) {
-  const validBlocks = tableNode.children.filter(node => {
-    const isBlock = node.type === 'BlockStatement';
-    if (!isBlock) {
-      return false;
+// For effective children, we skip over any if, unless, and else blocks
+function getEffectiveChildren(node) {
+  const unflattenedEffectiveChildren = AstNodeInfo.childrenFor(node).map(node => {
+    if (AstNodeInfo.isIf(node) || AstNodeInfo.isUnless(node) || AstNodeInfo.isEach(node)) {
```

Comment:
Mind adding `each-in`, `let`, and `with` support (along with tests) also?

Reply:
Added. Could you take another look?